src/normalizer.ts

```diff
@@ -145,6 +145,15 @@ rules.set('Transform const to singleton enum', schema => {
   }
 })
 
+rules.set('Spread required on allOf into its subSchema', schema => {
+  if (schema.allOf && schema.required) {
```

Comment:
We'll also want to consider:

- `anyOf` and `oneOf`
- Nested sub-schemas N layers deep